### PR TITLE
chore: Add Table of Contents to README

### PR DESCRIPTION
**Pull request by adarsh-jha-dev**
## What does this PR do?

This PR adds a Table of Contents (TOC) to the README of the Formbricks project. The TOC provides quick navigation links to different sections of the README, improving the overall accessibility and usability of the document. This change enhances the user experience and makes it easier for readers to find relevant information.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Chore (refactoring code, technical debt, workflow improvements)
- [ ] Enhancement (small improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change adds a new database migration
- [ ] This change requires a documentation update

## How should this be tested?

- Open the README file.
- Verify that a Table of Contents section is present with clickable links to different sections of the document.

### Required

- [x] Filled out the "How to test" section in this PR
- [x] Read [How we Code at Formbricks](<[https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md](https://formbricks.com/docs/contributing/how-we-code)>)
- [x] Self-reviewed my own code
- [ ] Commented on my code in hard-to-understand bits
- [ ] Ran `pnpm build`
- [ ] Checked for warnings, there are none
- [ ] Removed all `console.logs`
- [ ] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues
- [x] First PR at Formbricks? [Please sign the CLA!](https://formbricks.com/clmyhzfrymr4ko00hycsg1tvx) Without it we wont be able to merge it 🙏

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [x] Updated the Formbricks Docs if changes were necessary

### CODE REVIEW

**Comment by adarsh-jha-dev** [comment]
> The PR looks good overall. The addition of a Table of Contents is a great improvement for the readability of the README file. However, I have a minor suggestion to improve the formatting.
> 
> Currently, the Table of Contents is directly followed by the Features section. It would be better to add a brief introduction or overview of the product before jumping into the features. This would provide a smoother transition and give the reader a better understanding of what the product is about before diving into its features.
> 
> Here's a suggestion:
> 
> ```
> Formbricks is a powerful tool for creating in-product micro-surveys. It allows you to gather valuable insights from your users, enabling you to make data-driven decisions that enhance your product's user experience. With Formbricks, you can create surveys with our no-code editor, choose from a variety of templates, target specific user groups, and much more.
> 
> ### Table of Contents
> - [Features](#features)
> - [Getting Started](#getting-started)
>   - [Cloud Version](#cloud-version)
>   - [Self-hosted Version](#self-hosted-version)
>   - [Development](#development)
> - [Contribution](#contribution)
> - [License](#license)
> - [Security](#security)
> - [Contact](#contact)
> 
> ### Features
> ...
> ```
> 
> This suggestion is optional and depends on the style you prefer for your README. The current PR is already a great improvement to the documentation.

Done with this modification

**Comment by adarsh-jha-dev** [comment]
Can any of the maintainers please have a look at it and let me know if this can be merged ?

**Review by ShubhamPalriwala — CHANGES_REQUESTED** [review]
Hey @adarsh-jha-dev, thanks for opening the Pull Request! While I tried to test the PR, the hrefs do not look like they're working. 

Hint: The headings also have an emoji in front of them so please handle them 💪🏼

**Comment by adarsh-jha-dev** [comment]
> Hey @adarsh-jha-dev, thanks for opening the Pull Request! While I tried to test the PR, the hrefs do not look like they're working.
> 
> Hint: The headings also have an emoji in front of them so please handle them 💪🏼

I have done some changes, could you please review them and let me know if they are relevant  and working?

File: README.md [hunk 1]
```diff
@@ -14,10 +14,10 @@
...
-   <a href="https://news.ycombinator.com/item?id=32303986"><img src="https://img.shields.io/badge/Hacker%20News-122-%23FF6600" alt="Hacker News"></a>
-   <a href="https://www.producthunt.com/products/snoopforms"><img src="https://img.shields.io/badge/Product%20Hunt-%232%20Product%20of%20the%20Day-orange?logo=producthunt&logoColor=%23fff" alt="Product Hunt"></a>
-   <a href="https://github.blog/2023-04-12-github-accelerator-our-first-cohort-and-whats-next/"><img src="https://img.shields.io/badge/2023-blue?logo=github&label=Github%20Accelerator" alt="Github Accelerator"></a>
-<a href="https://github.com/formbricks/formbricks/issues?q=is:issue+is:open+label:%22%F0%9F%99%8B%F0%9F%8F%BB%E2%80%8D%E2%99%82%EF%B8%8Fhelp+wanted%22"><img src="https://img.shields.io/badge/Help%20Wanted-Contribute-blue"></a>
+   <a href="https://news.ycombinator.com/item?id=32303986"><img src="https://img.shields.io/badge/Hacker News-122-%23FF6600" alt="Hacker News"></a>
+   <a href="https://www.producthunt.com/products/snoopforms"><img src="https://img.shields.io/badge/Product Hunt-%232 Product of the Day-orange?logo=producthunt&logoColor=%23fff" alt="Product Hunt"></a>
```
Comment (jobenjada) [inline README.md:18]: all of these links are broken

File: README.md [hunk 2]
```diff
@@ -14,10 +14,10 @@
...
+   <a href="https://github.blog/2023-04-12-github-accelerator-our-first-cohort-and-whats-next/"><img src="https://img.shields.io/badge/2023-blue?logo=github&label=Github Accelerator" alt="Github Accelerator"></a>
```
Comment (jobenjada) [inline README.md:19]: same here

File: README.md [hunk 3]
```diff
@@ -14,10 +14,10 @@
...
+<a href="https://github.com/formbricks/formbricks/issues?q=is:issue+is:open+label:%22%F0%9F%99%8B%F0%9F%8F%BB%E2%80%8D%E2%99%82%EF%B8%8Fhelp+wanted%22"><img src="https://img.shields.io/badge/Help Wanted-Contribute-blue"></a>
```
Comment (jobenjada) [inline README.md:20]: and here

File: README.md [hunk 4]
```diff
@@ -31,7 +31,7 @@
 </p>
 
 <p align="center">
-  <a href="https://trendshift.io/repositories/2570" target="_blank"><img src="https://trendshift.io/api/badge/repositories/2570" alt="Trendshift Badge for formbricks/formbricks" style="width: 250px; height: 55px;" width="250" height="55"/></a>
+  <a href="https://trendshift.io/repositories/2570" target="_blank"><img src="https://trendshift.io/api/badge/repositories/2570" alt="Trendshift Badge for formbricks/formbricks" style="width: 250px; height="55px;" width="250" height="55"/></a>
```
Comment (jobenjada) [inline README.md:34]: is this intentional or a prettier issue?

File: README.md [hunk 5]
```diff
@@ -54,6 +54,19 @@ Formbricks is your go-to solution for in-product micro-surveys that will superch
 
 Formbricks helps you apply best practices from data-driven work and experience management to make better business decisions. Ask users as they experience your product - and leverage a significantly higher conversion rate. Gather all insights you can - including partial submissions and build conviction for the next product decision. Better data, better business.
 
+Formbricks is a powerful tool for creating in-product micro-surveys. It allows you to gather valuable insights from your users, enabling you to make data-driven decisions that enhance your product's user experience. With Formbricks, you can create surveys with our no-code editor, choose from a variety of templates, target specific user groups, and much more.
+
+### Table of Contents
```
Comment (jobenjada) [inline README.md:59]: this makes sense :) Please add a screenshot of what this would look like to the PR

**Review by jobenjada — CHANGES_REQUESTED** [review]
Thanks for your contribution! Changing this to draft until you fixed the issues flagged above :)

File: README.md (hunk 1 above)
Comment (adarsh-jha-dev) [reply README.md]: Corrected them

File: README.md (hunk 2 above)
Comment (adarsh-jha-dev) [reply README.md]: Corrected all the broken links

File: README.md (hunk 4 above)
Comment (adarsh-jha-dev) [reply README.md]: Was done mistakenly, reverted it back

**Comment by adarsh-jha-dev** [comment]
Screenshot 2023-10-28 202242


The screenshot as requested by @jobenjada

File: README.md (hunk 5 above)
Comment (adarsh-jha-dev) [reply README.md:59]: Added

**Comment by adarsh-jha-dev** [comment]
Hey @jobenjada , I have resolved the issues , I request you to please have a look at this once again

**Comment by adarsh-jha-dev** [comment]
> Hey @adarsh-jha-dev, thanks for opening the Pull Request! While I tried to test the PR, the hrefs do not look like they're working.
> 
> Hint: The headings also have an emoji in front of them so please handle them 💪🏼

Hey @ShubhamPalriwala , i have addressed your changes also , i have checked and made all the hrefs work now (handled the problem of the emojis), thanks for your assistance.

**Review by mattinannt — APPROVED** [review]
@adarsh-jha-dev thanks a lot for this change :-) Will merge it now!

**Review by jobenjada — APPROVED** [review]

**Comment by adarsh-jha-dev** [comment]
> @adarsh-jha-dev thanks a lot for this change :-) Will merge it now!

Thanks, should i merge the main with this to avoid any conflict?

**Review by ShubhamPalriwala — APPROVED** [review]

**Pull request merged**